src/hotspot/share/classfile/moduleEntry.hpp

```diff
@@ -219,39 +202,16 @@ class ModuleClosure: public StackObj {
 // PackageEntry node's correctly pointing at java.base's ModuleEntry.  No class
 // outside of java.base is allowed to be loaded pre-module system initialization.
 //
-// The ModuleEntryTable's lookup is lock free.
-//
-class ModuleEntryTable : public Hashtable<Symbol*, mtModule> {
-  friend class VMStructs;
-public:
-  enum Constants {
-    _moduletable_entry_size  = 109 // number of entries in module entry table
-  };
-
+class ModuleEntryTable : public CHeapObj<mtModule> {
 private:
   static ModuleEntry* _javabase_module;
-
-  ModuleEntry* new_entry(unsigned int hash, Handle module_handle, bool is_open,
-                         Symbol* name, Symbol* version, Symbol* location, ClassLoaderData* loader_data);
-  void add_entry(int index, ModuleEntry* new_entry);
-
-  int entry_size() const { return BasicHashtable<mtModule>::entry_size(); }
-
-  ModuleEntry** bucket_addr(int i) {
-    return (ModuleEntry**)Hashtable<Symbol*, mtModule>::bucket_addr(i);
-  }
-
-  static unsigned int compute_hash(Symbol* name) { return ((name == NULL) ? 0 : (unsigned int)(name->identity_hash())); }
-  int index_for(Symbol* name) const              { return hash_to_index(compute_hash(name)); }
+  ResourceHashtable<const Symbol*, ModuleEntry*, 109, ResourceObj::C_HEAP, mtModule,
+                    Symbol::compute_hash, Symbol::compare> _table;
```

Comment:
Is the `Symbol::compare` necessary? The `_invoke_method_type_table` table in systemDictionary.hpp is able to use the default comparison method.

Reply:
It is not necessary.